src/lib/Util/adapters/ibm/index.js

```diff
@@ -0,0 +1,266 @@
+    },
+    "entityRanges": [
+      {
+        "start": 13.02,
+        "end": 13.17,
+        "confidence": 0.68,
+        "text": "There",
+        "offset": 0,
+        "length": 5,
+        "key": "li6c6ld"
+      },
+      {
+        "start": 13.17,
+        "end": 13.38,
+        "confidence": 0.61,
+        "text": "is",
+        "offset": 6,
+        "length": 2,
+        "key": "pcgzkp6"
+      },
+      {
+        "start": 13.38,
+        "end": 13.44,
+        "confidence": 0.99,
+        "text": "a",
+        "offset": 9,
+        "length": 1,
+        "key": "ngomd9"
+      },
+      {
+        "start": 13.44,
+        "end": 13.86,
+        "confidence": 1,
+        "text": "day.",
+        "offset": 11,
+        "length": 4,
+        "key": "sgmfl4f"
+      }
+    ]
+  },
+  ...
+```
+ *
+ */
+
+const ibmToDraft = (ibmJson) => {
+
+  let x, y, z, topLvlResult, result;
+
+  const addPunctuation = true;
+  let newSentence = true;
+  let firstWordOfSentence = null;
+  let punctuatedWord = null;
+
+  // Watson doesn't include punctuation at this time (Dec-2018).
+  // Instead, manually capitalize first word after a "final" transcript block.
+  function capitalizeFirstLetter(string) {
+    if (!addPunctuation) {
+      return string;
+    } else {
+      return string.charAt(0).toUpperCase() + string.slice(1);
+    }
+  }
+
+  // Watson doesn't include punctuation at this time (Dec-2018).
+  // Instead, manually add punctuation on last word of "final" transcript blocks.
+  function punctuateEndOfSentence(firstWord, lastWord) {
+    if (!addPunctuation) {
+      return lastWord;
+
+    } else {
+
+      const interrogatives = [ 'when', 'where', 'why', 'how', 'does', "doesn't" ];
+
+      if (interrogatives.includes(firstWord)) {
+
+        return lastWord + '?';
+
+      } else {
+
+        return lastWord + '.';
+
+      }
+    }
+  }
+
+  const results = [];
+  for (x = 0; x < ibmJson.results.length; x++) { // result top level
+
+    topLvlResult = ibmJson.results[x];
+
+    for (y = 0; y < topLvlResult.results.length; y++) { // sentence level
+
+      const block = {};
+      const data = {};
+      const words = [];
+      const entityRanges = [];
+      const sentenceArray = [];
+      let offset = 0;
+
+      result = topLvlResult.results[y];
+
+      for (z = 0; z < result.alternatives[0].timestamps.length; z++) { // word level
+
+        const word = result.alternatives[0].timestamps[z][0];
+        const startTime = result.alternatives[0].timestamps[z][1];
+        const endTime = result.alternatives[0].timestamps[z][2];
+        const wordConfidence = result.alternatives[0].hasOwnProperty('word_confidence') ?
+          result.alternatives[0].word_confidence[z][1] : 100;
+
+        if (newSentence === true) {
+          punctuatedWord = capitalizeFirstLetter(word);
+          firstWordOfSentence = word;
+          newSentence = false;
+          data.speaker = 'TBC ' + topLvlResult.speaker_labels[z].speaker;
```

Comment:
IBM gives you speaker labels, and I've seen you are using that in the adapter, which is great
Only minor thing, I'd change the `TBC` to `Speaker`?

```suggestion
           data.speaker = 'Speaker ' + topLvlResult.speaker_labels[z].speaker;
```

As we have been (informally) using `TBC` as a place holder when the STT service doesn't return any speaker diarization info, and/or when it has not been incorporated in the adapter yet.

Reply:
Sounds great. I had originally had it as "Speaker" but didn't know if "TBC" had any... meaning and was trying to make this adapter 'fit in' as much as possible. ;-)